### PR TITLE
Fixed an issue with not being able to send events to initially started child actors when using `predictableActionArguments`

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/statelyai/xstate/issues/3548

### CODE REVIEW

File: packages/core/src/interpreter.ts [hunk 1]
```diff
@@ -810,7 +810,8 @@ export class Interpreter<
 
   private sendTo = (
     event: SCXML.Event<AnyEventObject>,
-    to: string | number | ActorRef<any>
+    to: string | number | ActorRef<any>,
+    immediate: boolean
```
Comment (Andarist) [inline packages/core/src/interpreter.ts:814]: The problem here was that executing initial actions is still done using `Interpreter#execute` - and doesn't happen from within `StateMachine#transition`:
https://github.com/statelyai/xstate/blob/0f1328c9c53c1409ee91ce00e99299643e054e6e/packages/core/src/interpreter.ts#L322-L336

So initial send actions were never sent out after my recent PR: https://github.com/statelyai/xstate/pull/3540

File: packages/core/src/interpreter.ts [hunk 2]
```diff
@@ -991,7 +992,7 @@ export class Interpreter<
           return;
         } else {
           if (sendAction.to) {
-            this.sendTo(sendAction._event, sendAction.to);
+            this.sendTo(sendAction._event, sendAction.to, _event === initEvent);
```
Comment (davidkpiano) [inline packages/core/src/interpreter.ts:995]: Was the problem that it was not sent immediately, or not sent at all?

File: packages/core/test/predictableExec.test.ts [hunk 3]
```diff
@@ -688,4 +688,39 @@ describe('predictableExec', () => {
...
+        },
+        done: {
+          type: 'final'
+        }
+      }
+    });
+
+    const service = interpret(machine).start();
+
+    expect(service.getSnapshot().value).toBe('done');
```
Comment (davidkpiano) [inline packages/core/test/predictableExec.test.ts:724]: In real-life scenarios, we can't assume that this is synchronous. Will the `machine` still get the `PONG` event _after_ it settles on the initial state (it should)? 

Maybe we can test this by instead listening for `.onDone(done)` so we don't depend on it being sync 🤔

File: packages/core/src/interpreter.ts (hunk 2 above)
Comment (Andarist) [reply packages/core/src/interpreter.ts:995]: they would be flushed within the next `service.send(ev)` - which could happen at any time, or not at all.

File: packages/core/test/predictableExec.test.ts (hunk 3 above)
Comment (Andarist) [reply packages/core/test/predictableExec.test.ts:724]: > In real-life scenarios, we can't assume that this is synchronous. Will the machine still get the PONG event after it settles on the initial state (it should)?

Yes, in this case, the test case is really more about the child receiving an event than about the parent receiving the response - I could slim it down.

> Maybe we can test this by instead listening for .onDone(done) so we don't depend on it being sync 🤔

I agree that those conceptually are async. However, I think that we also need to test synchronous logic as sync code often exposes more edge cases in scheduling etc. So it feels more beneficial to prefer sync tests in our test suite when possible - despite the mental model being async.

**Review by davidkpiano — APPROVED** [review]
